realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -711,4 +717,9 @@ public String toString() {
         sb.append("]");
         return sb.toString();
     }
+
+    @Override
+    protected Table getTable() {
+        return realm.schema.getTable(className);
```

Comment:
`getTable()` of `DynamicRealmObject` seems to work only if the instance is created by the constructor of `DynamicRealmObject(BaseRealm, String)`.

We should document this limitation or update to work with any constructor.